doc/mpich/TUNING_PARAMETERS

```diff
@@ -0,0 +1,98 @@
+This document discusses some of the new features added to the MPICH-based MPI
+implementation for the Aurora computer and the environment variables (CVARs)
+that the user can set to modify its behavior. MPICH is configured to use
+sensible default values for all the environment variables to provide overall
+good performance. However, users can change those defaults values to better tune
+their applications. This report provides documentation on how those CVARS work
+and the possible values that they can take. This documentation also provides
+information about per-communicator info hints that MPICH utilizes to make
+runtime decisions at a finer granularity.
+
+Notice that performance of the application depends on how ranks are mapped to
+the cores of the processor. For information on how to specify process-core
+binding with mpirun, information is provided here:
+https://wiki.mpich.org/mpich/index.php/Using_the_Hydra_Process_Manager#Process-core_Binding.
+
+
+
+This following section of this document provides information about threading
+optimizations (Section 1).
+
+**1. Multi-threading**
+
+
+MPI provides four different threading modes: MPI_THREAD_SINGLE,
+MPI_THREAD_FUNNELED, MPI_THREAD_SERIALIZED, and MPI_THREAD_MULTIPLE.
+MPI_THREAD_MULTIPLE is the most advanced threading mode and the focus of this
+Section. With this mode, an application runs with several threads, and any
+thread can make calls to MPI at any time. Thus, the MPI library needs to protect
+access to its internal data structures appropriately. It also tries to exploit
+concurrency by taking advantage of the availability of multiple hardware
+contexts in the network card, which we call VCIs (Virtual Communication
+Interfaces).
+
+MPICH provides two threading synchronization strategies. One strategy is to use
+a `global` coarse-grained lock to protect every MPI call that accesses shared
+state inside the MPI runtime. With this model, if two threads call concurrently
+the MPI runtime, the runtime will serialize the calls. Another strategy uses a
+fine-grained lock to protect the access to each network context or VCI. With
+this approach, threads trying to access the same VCI get serialized, but threads
+accessing different VCIs can execute in parallel. With the fine-grained locking
+model, MPICH supports a `direct` mode where the MPI runtime itself grabs a per
+VCI lock and a `lockless` mode where MPI does not grab a lock and relies on the
+network provider to provide fine grain synchronization (MPICH uses the
+FI_THREAD_SAFE mode from OFI to support the `lockless` mode).
```

Comment:
Document the configure option and runtime cvars.

Reply:
I'll add them.

Reply:
Done